### PR TITLE
PARQUET-782: Support writing to Arrow sinks

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

**Comment by xhochy** [comment]
Depends on https://github.com/apache/arrow/pull/215

File: src/parquet/arrow/reader.cc [hunk 1]
```diff
@@ -376,6 +376,79 @@ Status FlatColumnReader::Impl::TypedReadBatch(
 }
 
 template <>
+Status FlatColumnReader::Impl::TypedReadBatch<::arrow::BooleanType, BooleanType>(
+    int batch_size, std::shared_ptr<Array>* out) {
```
Comment (wesm) [inline src/parquet/arrow/reader.cc]: This is all because the `c_type` trait disappeared, or something else?

Comment (xhochy) [inline src/parquet/arrow/reader.cc]: Yes, that are all needed changes.

**Comment by wesm** [comment]
+1, thank you

**Pull request closed without merging**